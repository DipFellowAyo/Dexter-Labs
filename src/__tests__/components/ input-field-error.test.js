import React from 'react'
import Box from '@material-ui/core/Box'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

import InputFieldError from '../../components/forms/input-field-error'

describe('Input Field Error Component', () => {
  let inputFieldError

  beforeEach(() => {
    inputFieldError = shallow(<InputFieldError />)
  })
  afterAll(() => jest.resetAllMocks())

  it('should render without crashing', () => {
    expect(inputFieldError).toBeTruthy()
  })

  it('should match snapshots', () => {
    expect(toJson(inputFieldError)).toMatchSnapshot()
  })

  it('should render the error message', () => {
    inputFieldError.setProps({ errorText: 'error message' })
    expect(inputFieldError.text()).toEqual('error message')
  })

  it('should render a Box component', () => {
    expect(inputFieldError.find(Box)).toBeTruthy()
  })

  it('Box component accepts `pt` as props', () => {
    expect(inputFieldError.find(Box).props().pt).toEqual('2px')
  })

  it('Box component accepts `fontSize` as props', () => {
    expect(inputFieldError.find(Box).props().fontSize).toEqual('12px')
  })

  it('Box component accepts `color` as props', () => {
    expect(inputFieldError.find(Box).props().color).toEqual('red')
  })
})
