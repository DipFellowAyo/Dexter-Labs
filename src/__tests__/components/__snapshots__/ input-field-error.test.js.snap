// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field Error Component should match snapshots 1`] = `
<Styled(MuiBox)
  color="red"
  fontSize="12px"
  pt="2px"
/>
`;
